tuxemon/save.py

```diff
@@ -52,127 +54,174 @@
-        slot: The save slot to save the data to.
-
-    """
-    # Save a screenshot of the current frame
-    save_path = prepare.SAVE_PATH + str(slot) + ".save"
-    if prepare.SAVE_METHOD == "CBOR":
-        text = cbor.dumps(save_data)
-    else:
-        text = json.dumps(save_data, indent=4, separators=(",", ": "))
-    with open(save_path, "w") as f:
-        logger.info("Saving data to save file: " + save_path)
-        # Don't dump straight to the file: if we crash it would corrupt
-        # the save_data
-        f.write(text)
+	"""
+	Saves the current game state to a file using gzip compressed JSON.
+
+	Parameters:
+		save_data: The data to save.
+		slot: The save slot to save the data to.
+
+	"""
+	# Save a screenshot of the current frame
+	# if/else paradise
+	if config.compress_save:
+		if config.csave_lzma:
+			save_path = prepare.SAVE_PATH + str(slot) + ".csave.lza"
+			text = compress_json.dump(
+			save_data,
+			save_path,
+			compression_kwargs = {},
+			json_kwargs = { "indent": 4, 'separators': (",", ": ") }
+		)
+		else:
+			save_path = prepare.SAVE_PATH + str(slot) + ".csave.gz"
+			text = compress_json.dump(
+			save_data,
+			save_path,
+			compression_kwargs = {'compresslevel': 9},
+			json_kwargs = { "indent": 4, 'separators': (",", ": ") }
+		)
+	else:
+		save_path = prepare.SAVE_PATH + str(slot) + ".save"
+		if prepare.SAVE_METHOD == "CBOR":
+			text = cbor.dumps(save_data)
+		else:
+			text = json.dumps(save_data, indent=4, separators=(",", ": "))
+		with open(save_path, "w") as f:
+			logger.info("Saving data to save file: " + save_path)
+			# Don't dump straight to the file: if we crash it would corrupt
+			# the save_data
+			f.write(text)
+	#text = json.dumps(save_data, indent=4, separators=(",", ": "))
 
 
 def load(slot: int) -> Optional[Mapping[str, Any]]:
-    """
-    Loads game state data from a save file.
-
-    Parameters:
-        slot: The save slot to load game data from.
-
-    Returns:
-        Dictionary containing game data to load.
-
-    """
-    save_path = f"{prepare.SAVE_PATH}{slot}.save"
-    save_data = open_save_file(save_path)
-    if save_data:
-        return upgrade_save(save_data)
-    elif save_data is None:
-        # File not found; it probably wasn't ever created, so don't panic
-        return None
-    else:
-        save_data["error"] = "Save file corrupted"
-        save_data["player_name"] = "BROKEN SAVE!"
-        logger.error("Failed loading save file.")
-        return save_data
+	"""
+	Loads game state data from a save file.
+
+	Parameters:
+		slot: The save slot to load game data from.
+
+	Returns:
+		Dictionary containing game data to load.
+
+	"""
+	if config.compress_save:
+		if config.csave_lzma:
+			save_path = f"{prepare.SAVE_PATH}{slot}.csave.lza"
+		else:
+			save_path = f"{prepare.SAVE_PATH}{slot}.csave.gz"
+	else:
+		save_path = f"{prepare.SAVE_PATH}{slot}.save"
+
+	save_data = open_save_file(save_path)
+
+	# DEBUG STUFF HERE
```

Comment:
please remove the debug comments.  if its important, you can change to `logger.debug`